File: POO PHP/Aula12/Reptil.php
<?php

require_once 'Animal.php';

class Reptil extends Animal{

    //Atributo da Classe
    private $corEscama;
    
    //Metodo Acessor e Modificador
    public function getCorEscama() {
        return $this->corEscama;
    }
    public function setCorEscama($corEscama): void {
        $this->corEscama = $corEscama;
    }

    //Metodos Sobreposto de Animal
    public function alimentar() {
        echo "<p>Comendo vegetais no mato ...</p>";
    }
    public function emitirSom() {
        echo "<p>Fazendo som de Reptil ...</p>";
    }
    public function locomover() {
        echo "<p>Rastejando pelo chao ...</p>";
    }

}

File: POO JAVA/Aula12/src/aula12/Reptil.java
package aula12;

public class Reptil extends Animal{
    
    //Atributos da Classe
    private String corEscama;
    
    //Metodo Acessor e Modificador
    public String getCorEscama() {
        return corEscama;
    }
    public void setCorEscama(String corEscama) {
        this.corEscama = corEscama;
    }

    //Metodos de Sobreposição da Classe Aminal
    @Override
    public void locomover() {
        System.out.println("Rastejando ...");
    }
    @Override
    public void alimentar() {
        System.out.println("Comendo vegetais ...");
    }
    @Override
    public void emitirSom() {
        System.out.println("Emitindo som de Reptil.");
    }    
    
}
